fbw-a32nx/src/systems/fmgc/src/flightplanning/FlightPlanManager.ts

```diff
@@ -1921,8 +1921,12 @@ export class FlightPlanManager {
     public isWaypointInUse(icao: string): boolean {
         for (const fp of this._flightPlans) {
             for (let i = 0; i < fp?.waypoints.length; i++) {
-                if (fp.getWaypoint(i).icao === icao) {
-                    return true;
+                const wp = fp.getWaypoint(i);
+                if (wp != undefined)
+                {
+                    const waypointIcao =wp.icao;
+                        if (waypointIcao === icao)
+                            return true;
```

Comment:
Something is wrong if getWaypoint(i) is returning undefined... suggest looking into that if you're hitting an issue here.
```suggestion
                if (fp?.getWaypoint(i)?.icao === icao) {
                    return true;
```